### PR TITLE
additional required configuration

### PR DESCRIPTION
**Pull request by jschweda**
Add command line flag for additional required configuration and wait until file is available when starting the publisher

### CODE REVIEW

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Program.cs [hunk 1]
```diff
@@ -22,16 +22,41 @@ public static class Program {
...
+            // additional required configuration
+            var configFiles = args
+                .Select((v, i) => new { value = v, index = i })
+                .Where(
+                    i =>
+                        string.Compare(i.value.TrimStart('-'), "arc", StringComparison.InvariantCultureIgnoreCase) == 0 ||
+                        string.Compare(i.value.TrimStart('-'), "AdditionalRequiredConfiguration", StringComparison.InvariantCultureIgnoreCase) == 0);
+
+            foreach (var configFile in configFiles) {
+                // todo: read config from file
```
Comment (koepalex) [inline modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Program.cs]: Is this todo obsolete?

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Program.cs [hunk 2]
```diff
@@ -22,16 +22,41 @@ public static class Program {
         /// </summary>
         /// <param name="args"></param>
         public static void Main(string[] args) {
+            IConfigurationBuilder configBuilder = new ConfigurationBuilder()
+                   .SetBasePath(Directory.GetCurrentDirectory())
+                   .AddJsonFile("appsettings.json", optional: true)
+                   .AddEnvironmentVariables()
+                   .AddEnvironmentVariables(EnvironmentVariableTarget.User)
+                   .AddCommandLine(args)
+                   // making sure the standalone arguments are processed at last so they are not overriden
```
Comment (koepalex) [inline modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Program.cs:31]: Please remove empty line 32

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Program.cs (hunk 1 above)
Comment (jschweda) [reply modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Program.cs]: yes it was. thanks for spotting

**Comment by jschweda** [comment]
@koepalex please have another look

**Pull request closed without merging**